codec/types/interface_registry.go

```diff
@@ -129,11 +129,8 @@ type InterfaceRegistryOptions struct {
 	// ProtoFiles is the set of files to use for the registry. It is required.
 	ProtoFiles signing.ProtoFileResolver
 
-	// AddressCodec is the address codec to use for the registry. It is required.
-	AddressCodec address.Codec
-
-	// ValidatorAddressCodec is the validator address codec to use for the registry. It is required.
-	ValidatorAddressCodec address.Codec
+	// SigningOptions are the signing options to use for the registry.
+	SigningOptions signing.Options
```

Comment:
Still we just require a signing context to be passed in? Seems like signing options already takes proto files already so there's some redundancy here

Reply:
We could but `signing.Options` permits an nil `ProtoFileResolver` while interface registry does not.  What do you think we ought to do?

Reply:
Maybe just enforce the check at the interface registry level, or allow a nil resolver and default to hybrid resolver. But duplicated fields in both options structures might be confusing right?